src/policy/immix/immixspace.rs

```diff
@@ -197,14 +197,17 @@ impl<VM: VMBinding> ImmixSpace<VM> {
         &self.scheduler
     }
 
-    pub fn prepare(&mut self) {
-        // Update mark_state
-        if VM::VMObjectModel::LOCAL_MARK_BIT_SPEC.is_on_side() {
-            self.mark_state = Self::MARKED_STATE;
-        } else {
-            // For header metadata, we use cyclic mark bits.
-            unimplemented!("cyclic mark bits is not supported at the moment");
+    pub fn prepare(&mut self, major_gc: bool) {
```

Comment:
In generational immix, yes - immix as the mature space should not be called in nursery GCs. Immix space as a policy should be neutral to the plans and it may be used for a nursery GC. For example, in sticky immix, nursery collection is triggered for the immix space: https://github.com/JikesRVM/JikesRVM/blob/0b6002e7d746a829d56c90acfc4bb5c560faf634/MMTk/src/org/mmtk/plan/stickyimmix/StickyImmixCollector.java#L111